lib/browser/api/utility-process.ts

```diff
@@ -6,49 +6,46 @@ import { MessagePortMain } from '@electron/internal/browser/message-port-main';
 const { createProcessWrapper } = process._linkedBinding('electron_browser_utility_process');
 
 class IOReadable extends Readable {
-  _shouldPush: boolean = false;
-  _data: (Buffer | null)[] = [];
-  _resume: (() => void) | null = null;
-
-  _storeInternalData (chunk: Buffer | null, resume: (() => void) | null) {
-    this._resume = resume;
-    this._data.push(chunk);
-    this._pushInternalData();
+  #shouldPush: boolean = false;
+  #data: (Buffer | null)[] = [];
+  #resume: (() => void) | null = null;
+
+  storeInternalData (chunk: Buffer | null, resume: (() => void) | null) {
+    this.#resume = resume;
+    this.#data.push(chunk);
+    this.#pushInternalData();
   }
 
-  _pushInternalData () {
-    while (this._shouldPush && this._data.length > 0) {
-      const chunk = this._data.shift();
-      this._shouldPush = this.push(chunk);
+  #pushInternalData () {
+    while (this.#shouldPush && this.#data.length > 0) {
+      const chunk = this.#data.shift();
+      this.#shouldPush = this.push(chunk);
     }
-    if (this._shouldPush && this._resume) {
-      const resume = this._resume;
-      this._resume = null;
+    if (this.#shouldPush && this.#resume) {
+      const resume = this.#resume;
+      this.#resume = null;
       resume();
     }
   }
 
   _read () {
-    this._shouldPush = true;
-    this._pushInternalData();
+    this.#shouldPush = true;
+    this.#pushInternalData();
   }
 }
 
 export default class UtilityProcess extends EventEmitter {
-  _handle: any;
-  _stdout: IOReadable | null | undefined = new IOReadable();
-  _stderr: IOReadable | null | undefined = new IOReadable();
+  #handle: ElectronInternal.UtilityProcessWrapper | null;
+  #stdout: IOReadable | null | undefined = new IOReadable();
+  #stderr: IOReadable | null | undefined = new IOReadable();
   constructor (modulePath: string, args?: string[], options?: Electron.UtilityProcessConstructorOptions) {
     super();
-    let relativeEntryPath = null;
 
     if (!modulePath) {
       throw new Error('Missing UtilityProcess entry script.');
     }
 
-    if (process.resourcesPath) {
-      relativeEntryPath = path.relative(process.resourcesPath, modulePath);
-    }
+    const relativeEntryPath = path.relative(process.resourcesPath, modulePath);
 
     if (app.isPackaged && relativeEntryPath && relativeEntryPath.startsWith('..')) {
```

Comment:
```suggestion
    if (app.isPackaged && relativeEntryPath && relativeEntryPath.startsWith('..')) {
```

this test is technically wrong, it misfires for a filename `..foo.js`, for example. I don't _think_ `path.relative` can return a path like `./foo/bar/../../../../../../../etc/passwd` but this check wouldn't catch that either.

In general I very much mistrust this sort of path-string-based checking, because path strings are insanely complicated and it is very hard to catch all cases. For instance, does this behave well when the app is run from a network drive on Windows, where paths might look like `\\network.server\foo\bar\baz`? What if the network drive is also mounted at a drive letter like `Z:`, then `\\network.server\foo` is an alias for `Z:\foo`? What if there are symlinks involved, and `/foo/bar/baz` is a symlink to `/quux/baz`? Does this behave correctly when the filesystem is case-sensitive vs. case-insensitive? What about other crazy path and filesystem nonsense I haven't thought of?

I'd argue that we should drop this test unless we have a really clear idea of exactly what threat we're mitigating, and we are sure that the condition specifically mitigates that threat.

Reply:
I was trying to come up with a good example for your previous question in https://github.com/electron/electron/pull/34980#discussion_r981546385, rethinking my example with `--require` is not a valid comparison in the first place. Mitigation against `--require` injection was needed because external actors can inject untrusted scripts into main process or any other process through the environment variable `NODE_OPTIONS`. But that is not the case for `UtilityProcess` API, since an external actor must first compromise the main process or the mojo connection to do inject anything into this process.

Agreed with the path concerns, I will remove this path restriction for the `modulePath`.

Reply:
Refs https://github.com/electron/electron/pull/34980/commits/40a545b5da4496080527f11ec348eea7ce703197